packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx

```diff
@@ -1037,6 +1038,23 @@ const TinyMCE = (props: NoteBodyEditorProps, ref: any) => {
 			}
 		}
 
+		async function onCopy(event: any) {
+			const copiedContent = editor.selection.getContent();
+
+			// We need to remove extra url params from the image URLs while copying
+			// because some offline edtors do not show the image if there is
+			// an extra parameter in it's path.
+			// Related to - https://github.com/laurent22/joplin/issues/4602
+			const removeParametersFromUrl = (url: string) => {
+				const imageSrc = new URL(url);
+				return imageSrc.protocol === 'file:' ? imageSrc.origin + imageSrc.pathname : imageSrc.toString();
+			};
+
+			const updatedContent = HtmlUtils.replaceImageUrls(copiedContent, removeParametersFromUrl);
```

Comment:
Yes, put all these lines in a function under a file in TinyMCE/utils. From there, you can add a test unit next to the file. Search for ".test.ts" in the codebase for some examples.

Reply:
Done. Sorry for the delay, I got a bit busy.